src/Forms/GridField/GridField.php

```diff
@@ -1009,6 +1029,9 @@ public function gridFieldAlterAction($data, $form, HTTPRequest $request)
         }
 
         if ($request->getHeader('X-Pjax') === 'CurrentField') {
+            if ($this->getState(true)->Readonly) {
```

Comment:
`true` is the default value. This can just be `getState()`

Reply:
Done